### PR TITLE
Performance improvements for sending small images (tested with 64x64)

### PR DESCRIPTION
**Pull request by dnachbaur**
- directly compress them in the call thread
- keep JPEG out buffer allocated to save one memory allocation

### CODE REVIEW

File: deflect/ImageJpegCompressor.h [hunk 1]
```diff
@@ -71,6 +71,7 @@ class ImageJpegCompressor
 
 private:
     tjhandle _tjHandle;
+    std::vector<unsigned char> tjJpegBuf;
```
Comment (rdumusc) [inline deflect/ImageJpegCompressor.h]: _tjJpegBuf

File: deflect/StreamSendWorker.cpp [hunk 2]
```diff
@@ -128,6 +128,17 @@ Stream::Future StreamSendWorker::enqueueImage(const ImageWrapper& image,
         return promise.get_future();
     }
 
+    if (image.width <= 64 && image.height <= 64 &&
```
Comment (rdumusc) [inline deflect/StreamSendWorker.cpp]: #define 64 at the top

File: deflect/StreamSendWorker.cpp [hunk 3]
```diff
@@ -128,6 +128,17 @@ Stream::Future StreamSendWorker::enqueueImage(const ImageWrapper& image,
...
+        image.compressionPolicy == COMPRESSION_ON)
+    {
+        auto segment = _imageSegmenter.compressSingleSegment(image);
+        auto tasks = std::vector<Task>{
+            [this, segment] { return _sendSegment(segment); }};
+        if (finish)
+            tasks.emplace_back([this] { return _sendFinish(); });
+        return _enqueueRequest(std::move(tasks));
```
Comment (rdumusc) [inline deflect/StreamSendWorker.cpp]: these last 3 lines are the same as the normal case below, reuse them with an if/else for setting tasks?

**Review by rdumusc — COMMENTED** [review]
Good, a few minor details!

**Comment by dnachbaur** [comment]
updated

**Review by rdumusc — APPROVED** [review]

**Pull request merged**